modules/apps/transfer/ibc_module.go

```diff
@@ -352,11 +380,11 @@ func (IBCModule) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, pr
 // UnmarshalPacketData attempts to unmarshal the provided packet data bytes
 // into a FungibleTokenPacketData. This function implements the optional
 // PacketDataUnmarshaler interface required for ADR 008 support.
-func (IBCModule) UnmarshalPacketData(bz []byte) (interface{}, error) {
-	var packetData types.FungibleTokenPacketData
-	if err := json.Unmarshal(bz, &packetData); err != nil {
+func (im IBCModule) UnmarshalPacketData(bz []byte) (interface{}, error) {
```

Comment:
I would have assumed this returns the packet data based on the version, not that it converts to V2. I think it's fine if it converts to V2, but could be odd behaviour. It's technically okay since the interface only exists to extract from the memo in a standard way, which will be supported by V2 packet data in the exact same way as V1